proving/proving.go

```diff
@@ -150,7 +150,7 @@ func (p *Prover) loadMetadata() (*initialization.Metadata, error) {
 }
 
 func (p *Prover) verifyMetadata(m *Metadata) error {
-	if bytes.Compare(p.id, m.ID) != 0 {
+	if !bytes.Equal(p.id, m.ID) {
```

Comment:
is it a part of the fix or function was changed in 1.19?

Reply:
golangci-lint complained on my machine, that `bytes.Equal` should be used over `bytes.Compare` when possible. That's why I changed it; functionally it's identical and both functions have been available since Go 1.0